.github/workflows/publish-hotfix.yml

```diff
@@ -0,0 +1,182 @@
+# Description: this workflow publishes UI packages to `latest` tag on NPM.
+#
+# Triggered by: whenever a PR is merged to `hotfix` branch
+
+# If successful, we will publish all updated UI packages to
+#   - https://www.npmjs.com/package/@aws-amplify/ui
+#   - https://www.npmjs.com/package/@aws-amplify/ui-react
+#   - https://www.npmjs.com/package/@aws-amplify/ui-react-native
+#   - https://www.npmjs.com/package/@aws-amplify/ui-vue
+#   - https://www.npmjs.com/package/@aws-amplify/ui-angular
+
+name: Test and Publish / hotfix
+
+on:
+  push:
+    branches: [hotfix]
+
+permissions:
+  id-token: write # This is required for requesting the JWT
+  contents: write # Used to push tags to GitHub
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    outputs:
+      has-changesets: ${{ steps.has-changesets.outputs.has-changesets }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f # v3.4.0 https://github.com/actions/checkout/commit/24cb9080177205b6e8c946b17badbe402adc938f
+        with:
+          # This makes Actions fetch all Git history so that Changesets can generate changelogs with the correct commits
+          fetch-depth: 0
+      - name: Check if repo has unconsumed changesets
+        id: has-changesets
+        uses: ./.github/actions/has-changesets
+      # Note: if there are no changesets, that means that either
+      #         (1) A hot fix PR PR was just merged, or
+      #         (2) no commits with changesets have been merged after packages were last published.
+      #
+      #       In either way, we'll attempt to publish the packages. In case of (1), publish will succeed. In
+      #       case of (2), `@changesets/action` will know that UI packages have already published and will
+      #       skip publish.
+
+  publish-hotfix:
```

Comment:
Nit: you can use `reusable-tagged-publish`, e.g. https://github.com/aws-amplify/amplify-ui/blob/bc5533c145da3a048533813ad893a1b0d96ea341/.github/workflows/publish-next.yml#L152-L158

Reply:
Nice! updated only for hotfix, noticed the reusable script doesn't publish latest tag :)